src/ParallelAlgorithms/LinearSolver/CMakeLists.txt

```diff
@@ -8,9 +8,9 @@ add_library(${LIBRARY} INTERFACE)
 target_link_libraries(
   ${LIBRARY}
   INTERFACE
-  Blas
   Convergence
   DataStructures
+  LinearSolver
```

Comment:
I kept it specific to the linear solver so far because it's the inner product defined by a linear operator A. This is described in the documentation of `LinearSolver::inner_product`. How about we discuss how general we can make this in an issue, if you still think it should be more general?

Reply:
I'm fine leaving the question to another PR, but I'm now confused and need clarification: the `InnerProduct.hpp` moved to the directory `LinearSolver` in this PR appears to have only the interface function `inner_product` taking a `rhs` and a `lhs`, performing a simple dot product between them. Can you clarify what you mean by it being defined by an operator A? I would have anticipated that being something like  y^T A x.

Reply:
Yes you're right, I wasn't careful enough in the comment above. The inner product function only assumes that the `lhs` and `rhs` are elements of the same vector space, and that their c++ operators `*` and `+` implement the scalar multiplication and addition in that vector space.